### PR TITLE
chore: Bump web vault to v2023.7.1 and bump Rust

### PR DESCRIPTION
**Pull request by GeekCornerGH**
Requires #3592 for https://github.com/dani-garcia/vaultwarden/pull/3769/commits/2f594d4299574db622e500be0bb73a98637dc4ee

### CODE REVIEW

**Comment by BlackDex** [comment]
@GeekCornerGH, i see i forgot to change change the rust version number in the Dockerfile. Could you adjust that here in this pr? Thx!

**Comment by GeekCornerGH** [comment]
Sure @BlackDex, separate commit right?

**Comment by BlackDex** [comment]
Yea separate is fine, as its a different change.

**Comment by GeekCornerGH** [comment]
@BlackDex I'm not sure what Dockerfile i'm supposed to check, may you show me where?

**Comment by BlackDex** [comment]
@GeekCornerGH its regarding this: https://github.com/dani-garcia/vaultwarden/pull/3769/files#diff-4331345734d1b2a2578682b272eae8952289a0cffb0b5b6dd118f40802a4d055L5

**Comment by GeekCornerGH** [comment]
Got it, I was checking the source code to figure it out myself, then I saw your edit when writing this comment! Thanks.

**Comment by GeekCornerGH** [comment]
Renamed the PR title just to make it clearer it includes another commit in

**Comment by GeekCornerGH** [comment]
Rebased

File: Cargo.toml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name = "vaultwarden"
 version = "1.0.0"
 authors = ["Daniel García <dani-garcia@users.noreply.github.com>"]
 edition = "2021"
-rust-version = "1.69.0"
+rust-version = "1.71.1"
```
Comment (BlackDex) [inline Cargo.toml:6]: This should stay at `1.69.0`, as this is the MSRV, Minimal Stable Rust Version we will support.

**Review by BlackDex — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (GeekCornerGH) [reply Cargo.toml:6]: Oh sorry, my bad, I thought we had to bump this

**Review by BlackDex — APPROVED** [review]

**Pull request merged**